File: frontend/billing-frontend/src/app/shared/services/voice.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { VoiceRequest, VoiceResponse, VoiceAliasRequest } from '../models/models';
import { environment } from '../../../environments/environment';

@Injectable({
  providedIn: 'root'
})
export class VoiceService {
  private apiUrl = `${environment.apiUrl}/api/voice`;

  constructor(private http: HttpClient) {}

  processVoiceCommand(request: VoiceRequest): Observable<VoiceResponse> {
    return this.http.post<VoiceResponse>(`${this.apiUrl}/process`, request);
  }

  saveVoiceAlias(request: VoiceAliasRequest): Observable<any> {
    return this.http.post(`${this.apiUrl}/alias`, request);
  }
}

File: backend/src/main/java/com/example/billing/service/VoiceService.java
package com.example.billing.service;

import com.example.billing.dto.request.VoiceRequest;
import com.example.billing.dto.response.VoiceResponse;

public interface VoiceService {

    VoiceResponse processVoiceCommand(VoiceRequest request);

    void saveVoiceAlias(String spokenText, Long productId);
}
